Assemblers.Common/PackageReferenceProcessor.cs

```diff
@@ -300,11 +304,13 @@ private void ProcessResolvedPackages(IEnumerable<SourcePackageDependencyInfo> re
                             // Needs to be added as a reference in the dllImport attribute/script references.
                             nugetPackageAssemblies.DllImportNugetAssemblyReferences.Add(packageAssemblyReference);
 
-                            if (!resolvedPackage.Id.StartsWith(DevPackHelper.FilesPrefix))
+                            if (dontAddToPackageToInstall)
```

Comment:
I might be reading wrong, but this will always be false.    
dontAddToPackageToInstall  = false;
if ()
  set to true
else if()
  set to true
else()
 check condition
 
 if it reaches the else, it never execute the if or else if code. So it will always be false.